src/vs/vscode.proposed.d.ts

```diff
@@ -1181,6 +1181,45 @@ declare module 'vscode' {
 	}
 	//#endregion
 
+	/**
+	 * Class used to execute an extension callback as a task.
+	 */
+	export class CustomTaskExecution {
+		/**
+		 * @param callback The callback that will be called when the extension callback task is executed.
+		 */
+		constructor(callback: (terminalRenderer: TerminalRenderer, cancellationToken: CancellationToken, thisArg?: any) => Thenable<number | undefined>);
+
+		/**
+		 * The callback used to execute the task.
+		 */
+		callback: (terminalRenderer: TerminalRenderer, cancellationToken: CancellationToken, thisArg?: any) => Thenable<number | undefined>;
+	}
+
+	/**
+	 * A task to execute
+	 */
+	export class TaskWithCustomTaskExecution extends Task {
+		/**
+		 * Creates a new task.
+		 *
+		 * @param definition The task definition as defined in the taskDefinitions extension point.
+		 * @param scope Specifies the task's scope. It is either a global or a workspace task or a task for a specific workspace folder.
+		 * @param name The task's name. Is presented in the user interface.
+		 * @param source The task's source (e.g. 'gulp', 'npm', ...). Is presented in the user interface.
+		 * @param execution The process or shell execution.
+		 * @param problemMatchers the names of problem matchers to use, like '$tsc'
+		 *  or '$eslint'. Problem matchers can be contributed by an extension using
+		 *  the `problemMatchers` extension point.
+		 */
+		constructor(taskDefinition: TaskDefinition, scope: WorkspaceFolder | TaskScope.Global | TaskScope.Workspace, name: string, source: string, execution?: ProcessExecution | ShellExecution | CustomTaskExecution, problemMatchers?: string | string[]);
+
+		/**
+		 * The task's execution engine
+		 */
+		executionWithExtensionCallback?: ProcessExecution | ShellExecution | CustomTaskExecution;
```

Comment:
This is the override of execution?
````
		execution?: ProcessExecution | ShellExecution;
````
To make it clear that this something that will merge, let's call it execution2.